test/operators.jl

```diff
@@ -58,6 +58,25 @@ p = 1=>:foo
 @test_throws ArgumentError Base.scalarmax('a',['c','d'])
 @test_throws ArgumentError Base.scalarmax(['a','b'],'c')
 
+s1 = Set([1])
+s2 = Set([2])
+@test_throws MethodError min(s1, s2)
+@test_throws MethodError max(s1, s2)
+@test_throws MethodError minmax(s1, s2)
+
+struct TO
+  x
+end
+Base.isless(a::TO, b::TO) = isless(a.x, b.x)
+Base.isequal(a::TO, b::TO) = isequal(a.x, b.x)
+
+@test min(TO(1), TO(2)) == TO(1)
```

Comment:
If you wanted something where the fallbacks don't work as desired, you could define
```jl
Base.isless(a::TO23094, b::TO23094) = isless(a.x >> 1, b.x >> 1)
Base.isequal(a::TO23094, b::TO23094) = isequal(a.x >> 1, b.x >> 1)
```
I.e. every pair of integers are in the same `isequal` equivalence class.